cni/network/network.go

```diff
@@ -1008,83 +1062,88 @@ func (plugin *NetPlugin) Delete(args *cniSkel.CmdArgs) error {
 	// deleted, getNetworkName will return error of the type NetworkNotFoundError which will result in nil error as compliance
 	// with CNI SPEC as mentioned below.
 
-	numEndpointsToDelete := 1
-	// only get number of endpoints if it's multitenancy mode
-	if nwCfg.MultiTenancy {
-		numEndpointsToDelete = plugin.nm.GetNumEndpointsByContainerID(args.ContainerID)
+	// We get the network id and nw info here to preserve existing behavior
+	networkID, err = plugin.getNetworkID(args.Netns, nil, nwCfg)
+	if nwInfo, err = plugin.nm.GetNetworkInfo(networkID); err != nil {
+		if !nwCfg.MultiTenancy {
+			logger.Error("Failed to query network",
+				zap.String("network", networkID),
+				zap.Error(err))
+			// Log the error if the network is not found.
+			// if cni hits this, mostly state file would be missing and it can be reboot scenario where
+			// container runtime tries to delete and create pods which existed before reboot.
+			// this condition will not apply to stateless CNI since the network struct will be crated on each call
+			err = nil
+		}
 	}
-
-	logger.Info("Endpoints to be deleted", zap.Int("count", numEndpointsToDelete))
-	for i := 0; i < numEndpointsToDelete; i++ {
-		// Initialize values from network config.
-		networkID, err = plugin.getNetworkName(args.Netns, nil, nwCfg)
-		if err != nil {
-			// If error is not found error, then we ignore it, to comply with CNI SPEC.
-			if network.IsNetworkNotFoundError(err) {
-				err = nil
-				return err
-			}
-
-			logger.Error("Failed to extract network name from network config", zap.Error(err))
-			err = plugin.Errorf("Failed to extract network name from network config. error: %v", err)
+	// Initialize values from network config.
+	if err != nil {
+		// If error is not found error, then we ignore it, to comply with CNI SPEC.
+		if network.IsNetworkNotFoundError(err) {
+			err = nil
 			return err
 		}
```

Comment:
so networkid will be received from cns state right?

Reply:
network id won't be received from cns state, but hns network id will be. i believe this case (above) is only triggered if swiftv1 multitenancy and error from getting nwinfo